primap2/csg/_models.py

```diff
@@ -136,10 +266,18 @@ class StrategyDefinition:
 
     def find_strategy(self, fill_ts: xr.DataArray) -> FillingStrategyModel:
         """Find the strategy to use for the given filling timeseries."""
+        try:
+            return next(self.find_strategies(fill_ts))
+        except StopIteration:
+            raise KeyError(f"No matching strategy found for {fill_ts.coords}") from None
+
+    def find_strategies(
+        self, fill_ts: xr.DataArray
+    ) -> typing.Generator[FillingStrategyModel, None, None]:
+        """Yields all strategies to use for the timeseries, in configured order."""
         for selector, strategy in self.strategies:
-            if self.match(selector=selector, fill_ts=fill_ts):
-                return strategy
-        raise KeyError(f"No matching strategy found for {fill_ts.coords}")
+            if match_selector(selector=selector, ts=fill_ts):
+                yield strategy
```

Comment:
We use the python Generator protocol here with a co-routine.

Co-routines are special functions which `yield` instead of `returning. When you call `next(f)` on a function `f`, the function is run until it hits the next `yield`, then it is suspended and the yielded value is the return value of the `next` call. Instead of calling `next` explicitly, you can also iterate, e.g. with a for loop, which then will always run the function up to the next `yield` whenever it needs the next value in the loop. Turns out, this is exactly what we need for strategies.